src/Data/Vec/Properties.agda

```diff
@@ -76,13 +77,84 @@ unfold-take : ∀ n {m} x (xs : Vec A (n + m)) → take (suc n) (x ∷ xs) ≡ x
 unfold-take n x xs with splitAt n xs
 unfold-take n x .(xs ++ ys) | xs , ys , refl = refl
 
+take-distr-zipWith : ∀ {m n} → (f : A → B → C) →
+                     (u : Vec A (m + n)) → (v : Vec B (m + n)) →
+                     take m (zipWith f u v) ≡ zipWith f (take m u) (take m v)
+take-distr-zipWith {m = zero} f u v = refl
+take-distr-zipWith {m = suc m} f (u ∷ us) (v ∷ vs) = begin
+    take (suc m) (zipWith f (u ∷ us) (v ∷ vs))
+  ≡⟨⟩
+    take (suc m) (f u v ∷ (zipWith f us vs))
+  ≡⟨ unfold-take m (f u v) (zipWith f us vs) ⟩
+    f u v ∷ take m (zipWith f us vs)
+  ≡⟨ P.cong (f u v ∷_) (take-distr-zipWith f us vs) ⟩
+    f u v ∷ (zipWith f (take m us) (take m vs))
+  ≡⟨⟩
+    zipWith f (u ∷ (take m us)) (v ∷ (take m vs))
+  ≡⟨ P.cong₂ (zipWith f) (P.sym (unfold-take m u us)) (P.sym (unfold-take m v vs)) ⟩
```

Comment:
If you use `≡˘⟨` instead of `≡⟨` you can get rid of both of the `P.sym`s.

Reply:
Done